front/components/assistant/manager/TableTagSelector.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  Button,
+  ChevronDownIcon,
+  Chip,
+  DropdownMenu,
+  DropdownMenuCheckboxItem,
+  DropdownMenuContent,
+  DropdownMenuTrigger,
+  Spinner,
+} from "@dust-tt/sparkle";
+import { useState } from "react";
+
+import { useTheme } from "@app/components/sparkle/ThemeContext";
+import { useUpdateAgentTags } from "@app/lib/swr/tags";
+import type { WorkspaceType } from "@app/types";
+import type { TagType } from "@app/types/tag";
+
+type TableTagSelectorProps = {
+  tags: TagType[];
+  agentTags: TagType[];
+  agentConfigurationId: string;
+  owner: WorkspaceType;
+  onChange: () => Promise<any>;
+};
+
+export const TableTagSelector = ({
+  tags,
+  agentTags,
+  agentConfigurationId,
+  owner,
+  onChange,
+}: TableTagSelectorProps) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const { isDark } = useTheme();
+  const updateAgentTags = useUpdateAgentTags({
+    owner,
+    agentConfigurationId,
+  });
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        {agentTags.length === 0 ? (
+          <Button
+            variant="ghost"
+            size="xs"
+            label="Add tags"
+            isSelect
+            className="invisible text-muted-foreground group-hover:visible"
```

Comment:
NIT: need the darkmode class if you specify a color 🙏🏻  (same on the next button)